src/Donors/Routes/RegisterDonorRoutes.php

```diff
@@ -0,0 +1,156 @@
+<?php
+
+namespace Give\Donors\Routes;
+
+use Give\Donors\Controllers\DonorRequestController;
+use Give\Donors\ValueObjects\DonorRoute;
+use WP_REST_Request;
+use WP_REST_Server;
+
+/**
+ * @unreleased
+ */
+class RegisterDonorRoutes
+{
+    /**
+     * @var DonorRequestController
+     */
+    protected $donorRequestController;
+
+    /**
+     * @unreleased
+     */
+    public function __construct(DonorRequestController $donorRequestController)
+    {
+        $this->donorRequestController = $donorRequestController;
+    }
+
+    /**
+     * @unreleased
+     */
+    public function __invoke()
+    {
+        $this->registerGetDonor();
+        $this->registerGetDonors();
+    }
+
+    /**
+     * Get Donor route
+     *
+     * @unreleased
+     */
+    public function registerGetDonor()
+    {
+        register_rest_route(
+            DonorRoute::NAMESPACE,
+            DonorRoute::DONOR,
+            [
+                [
+                    'methods' => WP_REST_Server::READABLE,
+                    'callback' => function (WP_REST_Request $request) {
+                        return $this->donorRequestController->getDonor($request);
+                    },
+                    'permission_callback' => '__return_true',
+                ],
+                'args' => [
+                    'id' => [
+                        'type' => 'integer',
+                        'required' => true,
+                    ],
+                    'includeSensitiveData' => [
+                        'type' => 'boolean',
+                        'default' => false,
+                    ],
+                    'anonymousDonors' => [
+                        'type' => 'string',
+                        'default' => 'exclude',
+                        'enum' => [
+                            'exclude',
+                            'include',
+                            'redact',
+                        ],
+                    ],
+                ],
+            ]
+        );
+    }
+
+    /**
+     * Get Donors route
+     *
+     * @unreleased
+     */
+    public function registerGetDonors()
+    {
+        register_rest_route(
+            DonorRoute::NAMESPACE,
+            DonorRoute::DONORS,
+            [
+                [
+                    'methods' => WP_REST_Server::READABLE,
+                    'callback' => function (WP_REST_Request $request) {
+                        return $this->donorRequestController->getDonors($request);
+                    },
+                    'permission_callback' => '__return_true',
```

Comment:
We need to update the permission callbacks to check for the sensitive data parameters and authentication. If a client tries to get more data than they're authenticated to have, then we return false and the REST API will 403.

Reply:
Resolved in 0bc52d67abe9047987a915fc436ddab1cfbc91c4